ding_news.module

```diff
@@ -94,3 +94,30 @@ function ding_news_install_menu_position($op = 'install') {
     menu_position_delete_rule($exists);
   }
 }
+
+/**
+ * Implements hook_preprocess_views_view().
+ */
+function ding_news_preprocess_views_view(&$vars) {
+  $view = $vars['view'];
+  // Add a custom view more link for the news list for a group.
+  // The standard read more link points to standard news list and it does not
+  // seem to be possible to provide a custom url in another way.
+  // Adding a link in a footer text field means that the text will not be
+  // translatable and the url will not be processed using path alias.
+  if ($view->name = 'ding_news' &&
+    $view->current_display == 'ding_news_groups_list'
+  ) {
+    if (!empty($view->result[0]->og_membership_node_gid)) {
+      $vars['more'] = theme(
```

Comment:
Would it be possible to return a render array here and not calling the theme function directly. It would make overrides easier in theming layers.